lib/graphql/preload/instrument.rb

```diff
@@ -56,8 +64,21 @@ def instrument(_type, field)
         Promise.all(promises)
       end
 
-      private def preload_single_association(record, association)
-        GraphQL::Preload::Loader.for(record.class, association).load(record)
+      private def preload_single_association(record, association, scope)
+        # We would like to pass the `scope` (which is an
+        # `ActiveRecord::Relation`), directly into `Loader.for`, because that is
+        # what is needed for `Preloader.new`.  However, because the scope is
+        # created for each parent record, they are different objects and
+        # therefore would return different loaders, breaking batching.
+        # Therefore, we pass in `scope.to_sql`, which is the same for all the
+        # scopes and set the `scope` using an accessor.  So the actual scope
+        # object used will be the last one, which shouldn't make any difference,
+        # beacuse even though they are different objects, they are all
+        # equivalent.
+        loader = GraphQL::Preload::Loader.for(record.class, association,
+          scope.to_sql)
```

Comment:
`scope` may or may not be defined here, so we should at least wrap this in a `try`.

Reply:
you are totally correct.  Just ran into that myself.  I'll be updating shortly.